### PR TITLE
Remove `get` usage in geojson layer

### PR DESCRIPTION
**Pull request by Pessimistress**

### CODE REVIEW

**Comment by Pessimistress** [comment]
@ibgreen That's a good point. I agree that we should not just crash. Though there has been reports that we're being too forgiving and just quietly not rendering anything when the GeoJSON is malformed. Let me work on this a little more so we can emit meaningful errors when this happens.

**Comment by ibgreen** [comment]
Sounds good. 

For future reference, I've noticed that users have issues where nothing is rendering due to minor mistakes in the geojson, e.g. wrong nesting level on multipolygon arrays etc. 

It is hard for a new user to be confident about where things are broken when just starting out and nothing shows up. Some light error checking in the "parser" might be useful.

File: test/src/core-layers/geojson.spec.js [hunk 1]
```diff
@@ -23,27 +23,203 @@ import {
...
+  },
+  {
+    title: 'unknown geojson type',
+    argument: {type: 'Feature', geometry: {type: 'Something', coordinates: [0, 0]}},
+    error: /unknown geojson type/i
+  },
+  {
+    title: 'malformed geojson: Point',
+    argument: {type: 'Point'},
+    error: /coordinates is malformed/i
```
Comment (balthazar) [inline test/src/core-layers/geojson.spec.js]: should it be "coordinates are malformed" instead?

File: src/core-layers/geojson-layer/geojson.js [hunk 2]
```diff
@@ -39,25 +38,23 @@ export function getGeojsonFeatures(geojson) {
...
-    case 'MultiLineString':
-    case 'Polygon':
-    case 'MultiPolygon':
+  assert(geojson.type, 'GeoJSON does not have type');
+
+  switch (geojson.type) {
     case 'GeometryCollection':
-      // Wrap the geometry object in a 'Feature' object and wrap in an array
-      return [{type: 'Feature', properties: {}, geometry: geojson}];
+      assert(Array.isArray(geojson.geometries), 'GeoJSON does not have geometries');
```
Comment (1chandu) [inline src/core-layers/geojson-layer/geojson.js]: Super nit: should we specify we are expecting an `Array` type also ? may be the message should be `GeoJSON does not have geometries Array` ? Same for other asserts.

**Pull request merged**